tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/nonjaxrs/ValidLRAParticipant.java

```diff
@@ -0,0 +1,172 @@
+package org.eclipse.microprofile.lra.tck.participant.nonjaxrs;
+
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.Complete;
+import org.eclipse.microprofile.lra.annotation.Forget;
+import org.eclipse.microprofile.lra.annotation.ParticipantStatus;
+import org.eclipse.microprofile.lra.annotation.Status;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA.Type;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+
+import static org.eclipse.microprofile.lra.tck.participant.api.ParticipatingTckResource.ACCEPT_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.ParticipatingTckResource.RECOVERY_PARAM;
+
+/**
+ * TCK valid LRA participant containing a combination of valid participant method signatures
+ */
+@ApplicationScoped
+@Path(ValidLRAParticipant.RESOURCE_PATH)
+public class ValidLRAParticipant {
+    private static final Logger LOGGER = Logger.getLogger(ValidLRAParticipant.class.getName());
+    public static final String RESOURCE_PATH = "valid-nonjaxrs";
+    public static final String COMPLETED_COUNT_PATH = "completed";
+    public static final String COMPENSATED_COUNT_PATH = "compensated";
+    public static final String STATUS_COUNT_PATH = "status";
+    public static final String FORGET_COUNT_PATH = "forget";
+
+    private final AtomicInteger completedCount = new AtomicInteger(0);
+    private final AtomicInteger compensatedCount = new AtomicInteger(0);
+    private final AtomicInteger statusCount = new AtomicInteger(0);
+    private final AtomicInteger forgetCount = new AtomicInteger(0);
+
+    public static final String ENLIST_WITH_COMPLETE = "nonjaxrs-enlist-complete";
+    private int recoveryPasses;
+
+    @GET
+    @Path(ENLIST_WITH_COMPLETE)
+    @LRA(value = Type.REQUIRED)
+    public Response enlistWithComplete() {
+        return Response.ok().build();
+    }
+
+    public static final String ENLIST_WITH_COMPENSATE = "nonjaxrs-enlist-compensate";
+
+    @GET
+    @Path(ENLIST_WITH_COMPENSATE)
+    @LRA(value = Type.REQUIRED, cancelOn = Response.Status.INTERNAL_SERVER_ERROR)
+    public Response enlistWithCompensate() {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+
+
+    @Complete
+    public void completeWithException(URI lraId, URI parentId) {
+        verifyLRAId(lraId);
+
+        completedCount.incrementAndGet();
+
+        LOGGER.info(String.format("LRA id '%s' was completed", lraId));
```

Comment:
Logging at info (which is the default) will make the TCK runs very noisy - can you drop the level to debug. Similarly for the other info logging.